src/main/java/com/mineboundteam/minebound/container/AlloyFurnaceMenu.java

```diff
@@ -0,0 +1,161 @@
+    }
+    
+    public AlloyFurnaceMenu(int id, Inventory inventory, FriendlyByteBuf friendlyByteBuf) {
+        this(id, inventory, (AlloyFurnaceBlockEntity)inventory.player.level.getBlockEntity(friendlyByteBuf.readBlockPos()) , new SimpleContainerData(AlloyFurnaceBlockEntity.containerDataCount));
+    }
+
+    private void addHotBar(Inventory playerInventory) {
+        for (int i = 0; i < 9; ++i) {
+            this.addSlot(new Slot(playerInventory, i, 8 + i * 18, 144));
+        }
+    }
+
+    private void addInventory(Inventory playerInventory) {
+        for (int i = 0; i < 3; ++i) {
+            for (int j = 0; j < 9; ++j) {
+                this.addSlot(new Slot(playerInventory, j + i * 9 + 9, 8 + j * 18, 86 + i * 18));
+            }
+        }
+    }
+
+    public double getProgressPercent() {
+        return (double) containerData.get(0) / AlloyFurnaceBlockEntity.MAX_PROGRESS;
+    }
+
+    @Override
+    public @NotNull ItemStack quickMoveStack(@NotNull Player player, int index) {
+        // This code is from https://github.com/diesieben07/SevenCommons            - michael 11/25/2022
+        int vanillaSlotCount = 36;
+        int vanillaIndex = 0;
+        int teInventoryIndex = vanillaIndex + vanillaSlotCount;
+        Slot slot = slots.get(index);
+        if (slot == null || !slot.hasItem()) return ItemStack.EMPTY;
+        ItemStack sourceStack = slot.getItem();
+        ItemStack copyOfSourceStack = sourceStack.copy();
+        if (index < vanillaIndex + vanillaSlotCount) {
+            if (!moveItemStackTo(sourceStack, teInventoryIndex, teInventoryIndex
+                    + AlloyFurnaceBlockEntity.size, false)) {
+                return ItemStack.EMPTY;
+            }
+        } else if (index < teInventoryIndex + AlloyFurnaceBlockEntity.size) {
+            if (!moveItemStackTo(sourceStack, vanillaIndex, vanillaIndex + vanillaSlotCount, false)) {
+                return ItemStack.EMPTY;
+            }
+        } else {
+            System.out.println("Invalid slotIndex:" + index);
+            return ItemStack.EMPTY;
+        }
+        if (sourceStack.getCount() == 0) {
+            slot.set(ItemStack.EMPTY);
+        } else {
+            slot.setChanged();
+        }
+        slot.onTake(player, sourceStack);
+        return copyOfSourceStack;
+    }
+
+    @Override
+    public boolean stillValid(@NotNull Player player) {
+        return stillValid(ContainerLevelAccess.create(level, alloyFurnaceBlockEntity.getBlockPos()), player, BlockRegistry.ALLOY_FURNACE.get());
+    }
+
+    @Override
+    public void fillCraftSlotsStackedContents(StackedContents pItemHelper) {
+        // TODO Auto-generated method stub
+        
+    }
+
+    @Override
+    public void clearCraftingContent() {
+        // TODO Auto-generated method stub
+        
+    }
+
+    @Override
+    public boolean recipeMatches(Recipe<? super AlloyFurnaceMenu> pRecipe) {
+        // TODO Auto-generated method stub
+        return false;
+    }
```

Comment:
This function is defined twice in this here and throws an error, the second time is at the end of the file. Is this intentional or is one of them leftover and should be removed?